internal/provider/provider_framework.go

```diff
@@ -0,0 +1,502 @@
+						"gssapi": schema.ListNestedBlock{
+							Validators: []validator.List{
+								listvalidator.SizeAtMost(1),
+								listvalidator.ConflictsWith(
+									path.MatchRelative().AtParent().AtName("key_name"),
+									path.MatchRelative().AtParent().AtName("key_algorithm"),
+									path.MatchRelative().AtParent().AtName("key_algorithm"),
+								),
+							},
+							Description: "A `gssapi` block. Only one `gssapi` block may be in the configuration. " +
+								"Conflicts with use of `key_name`, `key_algorithm` and `key_secret`.",
+							NestedObject: schema.NestedBlockObject{
+								Attributes: map[string]schema.Attribute{
+									"realm": schema.StringAttribute{
+										Required:    true,
+										Description: "The Kerberos realm or Active Directory domain.",
+									},
+									"username": schema.StringAttribute{
+										Optional: true,
+										Description: "The name of the user to authenticate as. If not set the current " +
+											"user session will be used.",
+									},
+									"password": schema.StringAttribute{
+										Optional: true,
+										Validators: []validator.String{
+											stringvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("keytab")),
+											stringvalidator.AlsoRequires(path.MatchRelative().AtParent().AtName("username")),
+										},
+										Sensitive: true,
+										Description: "This or `keytab` is required if `username` is set. The matching " +
+											"password for `username`.",
+									},
+									"keytab": schema.StringAttribute{
+										Optional: true,
+										Validators: []validator.String{
+											stringvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("password")),
+											stringvalidator.AlsoRequires(path.MatchRelative().AtParent().AtName("username")),
+										},
+										Description: "This or `password` is required if `username` is set, not " +
+											"supported on Windows. The path to a keytab file containing a key for " +
+											"`username`.",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (p *dnsProvider) Configure(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
+	var providerConfig providerModel
+
+	var server, transport, timeout, keyname, keyalgo, keysecret, realm, username, password, keytab string
+	var port, retries int
+	var duration time.Duration
+	var gssapi bool
+
+	providerUpdateConfig := make([]providerUpdateModel, 1)
+	providerGssapiConfig := make([]providerGssapiModel, 1)
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &providerConfig)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !providerConfig.Update.IsNull() {
+		resp.Diagnostics.Append(providerConfig.Update.ElementsAs(ctx, &providerUpdateConfig, false)...)
+
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	if providerUpdateConfig[0].Server.IsNull() && len(os.Getenv("DNS_UPDATE_SERVER")) > 0 {
```

Comment:
I would've expected to need to do the equivalent of `(len(providerUpdateConfig) == 0 || providerUpdateConfig[0].Server.IsNull())` check before the `providerUpdateConfig[0]` access to prevent the potential panic, but not getting any panics not via the existing testing, so must be okay.